src/charm.py

```diff
@@ -121,6 +124,34 @@ def _is_legacy_v1_relation(self, relation):
 
         return True
 
+    def _configure_ingress(self, event: HookEvent) -> None:
+        """Set up ingress if a relation is joined, config changed, or a new leader election.
+
+        Also since :class:`TraefikRouteRequirer` may not have been constructed with an existing
+        relation if a :class:`RelationJoinedEvent` comes through during the charm lifecycle, if we
+        get one here, we should recreate it, but OF will give us grief about "two objects claiming
+        to be ...", so manipulate its private `_relation` variable instead.
+
+        Args:
+            event: a :class:`HookEvent` to signal a change we may need to respond to.
+        """
+        if not self.unit.is_leader():
+            return
+
+        # If it's a RelationJoinedEvent, set it in the ingress object
+        if isinstance(event, RelationJoinedEvent):
+            self._ingress._relation = event.relation
+
+        # No matter what, check readiness -- this blindly checks whether `ingress._relation` is not
+        # None, so it overlaps a little with the above, but works as expected on leader elections
+        # and config-change
+        if self._ingress.is_ready():
+            self._update_tracing_v1_relations()
```

Comment:
it doesn't make sense to update the tracing relations unless we've `submit_to_traefik` already in a previous event. 
First we submit to traefik, then traefik replies with 'here's your url', then we can update the tracing relations. 
So on ingress ready you want to update tracing v1 and v2 relations, but on ingress CREATED or joined, that's where you want to submit_to_traefik

Or maybe I'm misremembering the semantics of traefik_route.on.ready.

Please add a test for this :)

Reply:
You're correct - changed it to match what you said.